### DISCUSSION TITLE
Direct Javascript in blade problem

### DISCUSSION THREAD

**Discussion by ojsoft** [livewire/livewire#6927] [7. Bugs (archived)]
### Livewire version

3.0.5

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

Include any Javascript in a livewire blade of a livewire element that is used inside another liveries blade (nested).

### Example

Inside blade... of a nested livewire component (for easiness I use a jQuery example.)

```
<div id="text">Text</div>
<script>$('#text').hide()</script>
```

Text will be visible when the page is loaded. Of course this is just a stupid example. I do other things that cannot be easily done with alpinejs for example.

In Livewire 2 I could do things like that and they worked nicely.

In Livewire 3 I have to do it this way OUTSIDE of livewire. So yes I found a workaround, but what a pain for one simple line of javascript. I tried to get some things done with alpine js. Like putting the line in an alpinejs x-init. But this way this executes many times more than necessary and some things did not work at all. I am a heavy jQuery user I have to admit.

Can this functionality please be restored. Or alternatively maybe a livewire hook that gets called after the component is REALLY finished and ready for manipulation. The setTimeout version of this code is a hack really. 

```
<script>
  function myJS() {
      $('#text').hide();
  }

   document.addEventListener('livewire:init', () => {
      Livewire.hook('commit', ({ component, succeed }) => {
         succeed(() => {
            if (component.name === 'content-overlay') {
               setTimeout('myJS()', 200); // calling it directly won't work either
            }
         })
      })
   })
</script>
```

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by PhiloNL** [comment]
Try and use Alpine for this. Once the component is being rendered `x-init` will be invoked.

```
<div id="text" x-data x-init="$('#text').hide()">Text</div>
```

**Comment by ojsoft** [comment]
That's exactly the problem, is will be rendered each now and then with every change. I need something that is called one time when the element is loaded the first time and then not any more. a script injected did exactly that, but alpine does it on every change (that's what my debugging revealed at least).

The next thing is that is does not work evan. The code is executed yes, but the element seems not  to be not available at this time. Sounds weird, but this is what I experienced. I called a timeout to get around the problem, but the timeout then was called to many times by alpine on every change again that total chaos got generated, that's why I now call a timeout function outside on component load. This is still not perfect, but best solution I found.
